tools/debugging/channels_with_minimum_balance.py

```diff
@@ -0,0 +1,273 @@
+
+
+def channel_details(endpoint: str, token_address: str, partner: str) -> Optional[Dict]:
+    url_channel = f"{endpoint}/api/{API_VERSION}/channels/{token_address}/{partner}"
+    channel_response = requests.get(url_channel)
+
+    if not is_json_reponse(channel_response):
+        raise RuntimeError("Unexpected response from server, {channel_response}")
+
+    if channel_response.status_code == HTTPStatus.OK:
+        return channel_response.json()
+
+    return None
+
+
+def channel_deposit_if_necessary(channel_details: Dict, deposit: ChannelDeposit) -> None:
+    url_channel = (
+        f"{deposit.endpoint}/api/{API_VERSION}/"
+        f"channels/{deposit.token_address}/{deposit.partner}"
+    )
+
+    balance_current = int(channel_details["balance"])
+    balance_difference = deposit.minimum_capacity - balance_current
+    is_deposit_necessary = balance_difference > 0
+
+    if is_deposit_necessary:
+        deposit_current = int(channel_details["total_deposit"])
+        deposit_new = deposit_current + balance_difference
+        new_total_deposit = {"total_deposit": deposit_new}
+
+        log.info(f"Depositing to channel {deposit}")
+        response = requests.patch(url_channel, json=new_total_deposit)
+        if not is_successful_reponse(response):
+            raise RuntimeError(f"An error ocurrent while depositing to channel {deposit}")
+    else:
+        log.info(f"Channel exists and has enough capacity {deposit}")
+
+
+def channel_open_with_the_same_node(
+    channels_to_open: List[ChannelNew],
+    target_to_depositqueue: Dict[Tuple[str, str], JoinableQueue],
+) -> None:
+    """As of 0.100.5 channels cannot be open in parallel, starting multiple
+    opens at the same time can lead to the HTTP request timing out.  Therefore
+    here channels are open one after the other. (Issue #5446).
+    """
+    for channel_open in channels_to_open:
+        channel = channel_details(
+            channel_open.endpoint1, channel_open.token_address, channel_open.participant2
+        )
+
+        if channel is None:
+            channel_open_request = {
+                "token_address": channel_open.token_address,
+                "partner_address": channel_open.participant2,
+                "total_deposit": channel_open.minimum_capacity1,
+            }
+
+            log.info(f"Opening {channel_open}")
+            url_channel_open = f"{channel_open.endpoint1}/api/{API_VERSION}/channels"
+            response = requests.put(url_channel_open, json=channel_open_request)
+
+            if not is_successful_reponse(response):
+                raise RuntimeError(f"An error ocurrent while opening channel {channel_open}")
+
+        else:
+            deposit = ChannelDeposit(
+                channel_open.token_address,
+                channel_open.participant2,
+                channel_open.endpoint1,
+                channel_open.minimum_capacity1,
+            )
+            channel_deposit_if_necessary(channel, deposit)
```

Comment:
Why is this in the else branch? We want to make sure that we have sufficient deposit even when the channel has just been opened, right?

Reply:
channel opening does the deposit already. it is sent with the request [here](https://github.com/raiden-network/raiden/pull/5448/files#diff-34bdce61c76e505e38aac66139c45c5bR118), and processed by the REST API here https://github.com/raiden-network/raiden/blob/ec4e252a9d0db710d2ceae4a92fa886a9abd8933/raiden/api/rest.py#L681-L686